src/Aspire.Dashboard/Otlp/Storage/TelemetryRepository.cs

```diff
@@ -271,6 +274,11 @@ private void RaiseSubscriptionChanged(List<Subscription> subscriptions)
 
     public void AddLogs(AddContext context, RepeatedField<ResourceLogs> resourceLogs)
     {
+        if (_pauseManager.StructuredLogsPaused)
+        {
+            return;
```

Comment:
Could you add a unit test to the telemetry repository tests that tests pausing? I think just one test where everything is paused, and then you attempt to write the 3 kinds of telemetry, and then check that there is no new data.